reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt

```diff
@@ -14,6 +14,14 @@ expect open class SerialDisposable() : Disposable {
      */
     override fun dispose()
 
+    /**
+     * Atomically removes any existing [Disposable] and disposes this [SerialDisposable].
+     * Does not dispose the removed [Disposable].
+     *
+     * @return the removed [Disposable] if any.
+     */
+    internal fun clearAndDispose(): Disposable?
```

Comment:
Can be implemented as an extension function to not pollute public class with internal functions.
```kotlin
val d = replace(null)
dispose()
return d
```

Reply:
The idea is to perform the operation under one `synchronized` block.